driver-sync/src/test/functional/com/mongodb/client/unified/Entities.java

```diff
@@ -529,30 +522,42 @@ private void initClient(final BsonDocument entity, final String id,
                                 ServerMonitoringModeUtil.fromString(value.asString().getValue())));
                         break;
                     case "authMechanism":
-                        if (value.equals(new BsonString(AuthenticationMechanism.MONGODB_OIDC.getMechanismName()))) {
-                            clientSettingsBuilder.credential(MongoCredential.createOidcCredential(null));
+                        if (value.equals(new BsonString(MONGODB_OIDC.getMechanismName()))) {
+                            // authMechanismProperties depends on authMechanism
+                            BsonDocument authMechanismProperties = entity
+                                    .getDocument("uriOptions")
+                                    .getDocument("authMechanismProperties");
+                            boolean hasPlaceholder = authMechanismProperties.equals(
+                                    new BsonDocument("$$placeholder", new BsonInt32(1)));
+                            if (!hasPlaceholder) {
+                                throw new UnsupportedOperationException("Unsupported authMechanism: " + value);
+                            }
+
+                            String env = getenv("OIDC_ENV");
+                            if (env == null) {
+                                env = "test";
+                            }
```

Comment:
`env` is not expected to be `null`. We should assert that `env` is not `null`, instead of using `test` when this happens. This way we also won't accidentally run all tests using the `"test"` `ENVIRONMENT`.